packages/strapi-admin/admin/src/app.js

```diff
@@ -195,3 +196,10 @@ export { dispatch };
 if (window.Cypress) {
   window.__store__ = Object.assign(window.__store__ || {}, { store });
 }
+
+// Install ServiceWorker and AppCache in the end since
+// it's not most important operation and if main code fails,
+// we do not want it installed
+if (NODE_ENV === 'production') {
+  require('offline-plugin/runtime').install(); // eslint-disable-line global-require
```

Comment:
adding service workers is kinda risky as of now it is not well known and used. we should be careful about that one.